### PR TITLE
Update CCTP Lib dependency with static Bzip2 lib

### PR DESCRIPTION
**Pull request by ptagl**
This PR updates the CCTP Lib dependency to the latest version, that includes a small fix to statically link the Bzip2 library.
Such change is required to avoid errors when compiling Zend on systems on which Bzip2 is installed as shared library, since in that case Cargo uses the dynamic linking without providing the Bzip2 symbols.

Such update required also some small changes to make unit tests compatible with the new CCTP Lib.

### CODE REVIEW

**Comment by ptagl** [comment]
This PR should be merged after

- https://github.com/HorizenOfficial/zendoo-cctp-lib/pull/41
- https://github.com/HorizenOfficial/zendoo-cctp-lib/pull/42

After having merged the CCTP Lib and releasing a new version, this branch should be changed accordingly to include the right version as dependency.

File: examples/sampleCalls.cpp
Comment (DanieleDiBenedetto) [inline examples/sampleCalls.cpp:1571]: This is the test not passing right ? We can remove it I think

File: src/lib.rs
Comment (DanieleDiBenedetto) [inline src/lib.rs:1480]: Can't we re-use some code among the zendoo_batch_verify_proofs_by_id_with_segment_size and zendoo_batch_verify_proofs_by_id like you did for the internal functions ?

**Review by DanieleDiBenedetto — CHANGES_REQUESTED** [review]

File: Cargo.lock
Comment (ptagl) [reply Cargo.lock:11]: I tested the Zend compilation through a temporary "integration" branch, I had to update some crate dependencies but in the end it works.

**Comment by ptagl** [comment]
This PR should be put on hold since we discovered some weird behavior while developing the requested changes. In particular, we saw some tests in `mcTest` fail when using Darlin proving system (instead of Coboundary Marlin that is working as expected).

The changes contained in this PR are made to be compatible with the latest version of the CCTP Lib, that introduced a new way of handling the DLog keys: it's not possible to call `init_dlog_keys()` multiple times anymore, during the execution of the application it's only possible to call it once and then using trimmed keys whenever we want to generate or verify proofs with a smaller segment size.

On Zend, we rely on such mechanism because the main application always uses a segment size of 2^18 (and it's only used for verifying proofs since they are generated by Sidechains) but we needed to implemented fast Python tests to generate and verify proofs with a segment size of 2^9 (to make it faster).

With the old version of MC Crypto Lib (and of CCTP Lib) we used the special function `init_dlog_keys_test_mode(max_segment_size, supported_segment_size)` to initialize keys with a smaller supported segment size (2^9) to generate proofs that were compatible with the verification process of Zend main application (that uses 2^18 as a max segment size) .

Since such function is not available anymore, we can only generate keys using `init_dlog_keys(segment_size)` and relying on trimming the DLog keys when generating and verifying the proofs. So, basically, the current situation is:

- Zend uses DLog keys with a segment size of 2^18 (for verification only, there is no need to generate proofs)
- mcTestCall initializes DLog keys with a segment size of 2^9 and contextually generates the proving and verification keys with the same parameter
- mcTestCall uses DLog keys with a segment size of 2^18 when generating and verifying proofs, but now uses an additional parameter `segment_size` when calling `create_proof()` and `verify_proof()` to trim the DLog keys

Everything is perfectly compatible when using Coboundary Marlin, but it immediately stops working for Darlin because it seems like we cannot use DLog keys of 2^9 segment size to generate the proving and verification keys and then use 2^18 when generating the proof (even though the keys get trimmed). This behavior must be analyzed carefully to avoid to introduce incompatibilities in Zendoo and eventually hidden hard forks. Moreover, it should be checked that the Sidechain SDK (which uses the updated version of CCTP Lib through the SC Crypto Lib) has not any inconsistency (it should be generally safe since it's not currently using Darlin).

**Comment by DanieleDiBenedetto** [comment]
Issues were addressed and are now part of https://github.com/HorizenOfficial/zendoo-mc-cryptolib/pull/60

**Pull request closed without merging**
